pkg/azure/securitygroups.go

```diff
@@ -0,0 +1,231 @@
+// TODO Make this private once gwdeployer is done
+
+func (c *CloudInfo) OpenGWPorts(infraID string, ports []api.PortSpec,
+	sgClient *network.SecurityGroupsClient, networkInterfaces *[]network.Interface) error {
+	groupName := infraID + externalSecurityGroupSuffix
+
+	isFound := checkIfSecurityGroupPresent(groupName, sgClient, c.BaseGroupName)
+	if isFound {
+		return nil
+	}
+
+	securityRules := []network.SecurityRule{}
+	for i, port := range ports {
+		securityRules = append(securityRules, c.createSecurityRule(externalSecurityRulePrefix, port.Protocol,
+			port.Port, int32(baseExternalInternal+i), network.SecurityRuleDirectionInbound),
+			c.createSecurityRule(externalSecurityRulePrefix, port.Protocol, port.Port,
+				int32(baseExternalInternal+i), network.SecurityRuleDirectionOutbound))
+	}
+
+	nwSecurityGroup := network.SecurityGroup{
+		Name:     &groupName,
+		Location: to.StringPtr(c.Region),
+		SecurityGroupPropertiesFormat: &network.SecurityGroupPropertiesFormat{
+			SecurityRules:     &securityRules,
+			NetworkInterfaces: networkInterfaces,
+		},
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 300*time.Second)
+	defer cancel()
+
+	future, err := sgClient.CreateOrUpdate(ctx, c.BaseGroupName, groupName, nwSecurityGroup)
+	if err != nil {
+		return errors.Wrapf(err, "creating security group %q failed", groupName)
+	}
+
+	err = future.WaitForCompletionRef(ctx, sgClient.Client)
+
+	return errors.Wrapf(err, "Error creating  security group %v ", groupName)
+}
+
+// TODO Make this private once gwdeployer is done
+
+func (c *CloudInfo) RemoveGWFirewallRules(infraID string, sgClient *network.SecurityGroupsClient) error {
+	groupName := infraID + externalSecurityGroupSuffix
+
+	isFound := !checkIfSecurityGroupPresent(groupName, sgClient, c.BaseGroupName)
+	if isFound {
+		return nil
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 300*time.Second)
+	defer cancel()
+
+	nwSecurityGroup, err := sgClient.Get(ctx, c.BaseGroupName, groupName, "")
+	if err != nil {
+		return errors.Wrapf(err, "error getting the submariner gateway securitygroup %q", groupName)
+	}
+
+	nwSecurityGroup.SecurityGroupPropertiesFormat.NetworkInterfaces = nil
+
+	updateFuture, err := sgClient.CreateOrUpdate(ctx, c.BaseGroupName, groupName, nwSecurityGroup)
+	if err != nil {
+		return errors.Wrapf(err, "removing security group %q from interfaces failed", groupName)
+	}
+
+	err = updateFuture.WaitForCompletionRef(ctx, sgClient.Client)
+
+	if err != nil {
+		return errors.Wrapf(err, "waiting for  the submariner gateway security group  %q to be updated failed", groupName)
+	}
+
+	deleteFuture, err := sgClient.Delete(ctx, c.BaseGroupName, groupName)
+	if err != nil {
+		return errors.Wrapf(err, "deleting security group %q failed", groupName)
+	}
+
+	err = deleteFuture.WaitForCompletionRef(ctx, sgClient.Client)
+
+	if err != nil {
+		return errors.Wrapf(err, "waiting for s the submariner gateway  ecurity group  %q to be deleted failed", groupName)
```

Comment:
`waiting for s the submariner gateway ecurity group` -> `waiting for the submariner gateway security group`

Reply:
Done.